alacritty/src/config/window.rs

```diff
@@ -74,10 +74,18 @@ impl Default for WindowConfig {
 impl WindowConfig {
     #[inline]
     pub fn dimensions(&self) -> Option<Dimensions> {
-        if self.dimensions.columns.0 != 0
-            && self.dimensions.lines != 0
-            && self.startup_mode != StartupMode::Maximized
-        {
+        let is_columns_set = self.dimensions.columns.0 != 0;
+        let is_lines_set = self.dimensions.lines != 0;
+
+        // Throw warning if only one of columns, lines is set
+        if is_lines_set ^ is_columns_set {
+            warn!(
+                target: LOG_TARGET_CONFIG,
+                "Both lines and columns must be set for window.dimensions to take effect"
```

Comment:
The message should instruct the user more properly on what is wrong, not on the invariant we're trying to ensure. It should state that one of the dimensions is zero while the other one is set, not claim that one is not set.